app/types/s3_access.py

```diff
@@ -0,0 +1,111 @@
+from datetime import UTC, datetime, timedelta
+from io import BytesIO
+from logging import Logger
+
+import boto3
+import botocore
+import botocore.exceptions
+
+from app.core.utils.config import Settings
+from app.types.exceptions import (
+    InvalidS3AccessError,
+    InvalidS3BucketNameError,
+    InvalidS3FileNameError,
+)
+
+
+class S3Access:
+    """Class to manage S3 access with object locking."""
+
+    def __init__(
+        self,
+        settings: Settings,
+        failure_logger: Logger,
+        folder: str = "logs/",
+        retention: int = -1,
+    ) -> None:

```

Comment:
Could you add a docstring explaining required parameters. Especially `retention`